src/CollectionDataProvider.php

```diff
@@ -2,194 +2,87 @@
 
-use ApiPlatform\Core\Api\FilterLocatorTrait;
-use ApiPlatform\Core\DataProvider\CollectionDataProviderInterface;
-use ApiPlatform\Core\Exception\ResourceClassNotSupportedException;
-use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\State\Pagination\PaginationOptions;
+use ApiPlatform\State\Pagination\PartialPaginatorInterface;
+use ApiPlatform\State\ProviderInterface;
 use Aura\SqlQuery\Common\SelectInterface;
 use CCMBenchmark\Ting\ApiPlatform\Filter\FilterInterface;
-use CCMBenchmark\Ting\MetadataRepository;
+use CCMBenchmark\Ting\ApiPlatform\Pagination\PaginationConfig;
+use CCMBenchmark\Ting\ApiPlatform\Pagination\Paginator;
 use CCMBenchmark\Ting\Repository\HydratorSingleObject;
 use CCMBenchmark\Ting\Repository\Repository;
-use Psr\Container\ContainerInterface;
-use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\DependencyInjection\ServiceLocator;
 use Symfony\Component\HttpFoundation\RequestStack;
 
 use function array_column;
-use function dump;
-use function implode;
-use function is_array;
-use function is_string;
-use function sprintf;
-use function str_replace;
 
-class CollectionDataProvider implements CollectionDataProviderInterface
+class CollectionDataProvider implements ProviderInterface
 {
-    use FilterLocatorTrait;
-
-    const PAGE_PARAMETER_NAME_DEFAULT = 'page';
-
-    /**
-     * @var RepositoryProvider
-     */
-    private $repositoryProvider;
-    /**
-     * @var MetadataRepository
-     */
-    private $metadataRepository;
-    /**
-     * @var RequestStack
-     */
-    private $requestStack;
-    private $pagination;
-    private $order;
-
     public function __construct(
-        RepositoryProvider $repositoryProvider,
-        ContainerInterface $filterLocator,
-        ResourceMetadataFactoryInterface $resourceMetadataFactory,
-        RequestStack $requestStack,
-        array $pagination,
-        array $order
+        private RepositoryProvider $repositoryProvider,
+        private RequestStack $requestStack,
+        private PaginationOptions $paginationOptions,
+        private ServiceLocator $filterLocator,
     ) {
-        $this->repositoryProvider = $repositoryProvider;
-        $this->resourceMetadataFactory = $resourceMetadataFactory;
-        $this->requestStack = $requestStack;
-        $this->pagination = $pagination;
-        $this->order = $order;
-        $this->setFilterLocator($filterLocator);
     }
 
     /**
      * {@inheritdoc}
      */
-    public function getCollection(string $resourceClass, string $operationName = null)
+    public function provide(Operation $operation, array $uriVariables = [], array $context = []): ?PartialPaginatorInterface
     {
+        $resourceClass = $operation->getClass() ?? '';
+        $operationName = $operation->getName();
+
         $request = $this->requestStack->getCurrentRequest();
         if (null === $request) {
```

Comment:
```suggestion
        if (null === $request || $resourceClass === '') {
```
Il peut être intéressant de faire une early return si on a n'a pas de resource identifiée non ?